docs/pages/developer_guide.md

```diff
@@ -78,7 +83,9 @@ There are tools for your favorite IDE to automatically format your code. Example
 - [vim](https://github.com/rhysd/vim-clang-format)
 - [and many more](https://clang.llvm.org/docs/ClangFormat.html)
 
-Before committing any changes, you can run the following
+Before formatting the code with clang-format, ensure that you have latest clang-format version by running `clang-format --version`
```

Comment:
Add: To upgrade, on macOS, run `brew upgrade clang-format`, on Ubuntu, run `sudo apt upgrade clang-format`.